mobile/library/objective-c/EnvoyEngineImpl.mm

```diff
@@ -414,18 +405,24 @@ - (instancetype)initWithRunningCallback:(nullable void (^)())onEngineRunning
     };
   }
 
-  // TODO(Augustyniak): Everything here leaks, but it's all tied to the life of the engine.
-  // This will need to be updated for https://github.com/envoyproxy/envoy-mobile/issues/332.
-  envoy_event_tracker native_event_tracker = {NULL, NULL};
+  std::unique_ptr<Envoy::EnvoyEventTracker> native_event_tracker =
+      std::make_unique<Envoy::EnvoyEventTracker>();
   if (eventTracker) {
-    EnvoyEventTracker *objcEventTracker =
-        [[EnvoyEventTracker alloc] initWithEventTrackingClosure:eventTracker];
-    native_event_tracker.track = ios_track_event;
-    native_event_tracker.context = CFBridgingRetain(objcEventTracker);
+    native_event_tracker->on_track =
+        [eventTracker =
+             std::move(eventTracker)](const absl::flat_hash_map<std::string, std::string> &events) {
+          NSMutableDictionary *newMap = [NSMutableDictionary new];
+          for (const auto &[cppKey, cppValue] : events) {
+            NSString *key = @(cppKey.c_str());
+            NSString *value = @(cppValue.c_str());
+            newMap[key] = value;
```

Comment:
why do we need this?

Reply:
That's to replace `to_ios_map(map)` since we can use native C++ map instead of `envoy_map`.